File: cntrea/27/5.py
nums = list(map(int, input().split()))
n = len(nums)



table = [[None for l in range(13)] for k in range(5)]
for k in range(5):
    for l in (1, 3, 4, 12):
        for i in range(n):
            if (nums[i] % 5 == k) and (nums[i] % l == 0):
                if table[k][l] == None:
                    table[k][l] = i
                else:
                    if nums[i] > nums[table[k][l]]:
                        table[k][l] = i

res = None
for i in range(n):
    for k in range(5):
        for l in (1, 3, 4, 12):
            j = table[k][l]
            if (j != None) and (j != i):
                if (nums[i] * nums[j] % 12 == 0) and ((nums[i] + nums[j]) % 5 == 0):
                    if res == None:
                        res = nums[i] + nums[j]
                    else:
                        res = max(res, nums[i] + nums[j])


print((res if res != None else "NO"))

File: olympics/vsosh/5.cpp
#pragma GCC optimize("O3")

#include <bits/stdc++.h>

using namespace std::chrono;
using namespace std;

#define ll long long
#define dbl double
#define ldbl long double
#define uint unsigned int
#define peque priority_queue
#define map unordered_map

int main(){
    ios_base::sync_with_stdio(0);
    cin.tie(0);

    vector<set<int>> display{{0, 1, 3, 5, 7, 8}, {2, 3, 7}, {0, 3, 6, 8}, {0, 2, 4, 6}, {1, 3, 4, 7}, {0, 1, 4, 7, 8}, {2, 4, 5, 7, 8}, {0, 2, 5}, {0, 1, 3, 4, 5, 7, 8}, {0, 1, 3, 4, 6}};
   
    vector <bool> number(10, true);
    vector<vector<bool>> table(10, number);

    for (int i = 0; i < 10; i++){
        for (auto a : display[i]){
            table[i][a] = false;
        }
    }

    int n;
    cin >> n;
    
    vector <int> nums(n);
    for (int i = 0; i < n; i++){
        cin >> nums[i];
    }

    for (int x : nums){
        for (auto &s : table){
            for (auto a : display[x]){
                s[a] = true;
            }
        }
    }

    vector <int> res;
    for (int i = 0; i < 10; i++){
        bool f = 1;
        for (auto a : table[i]){
            f = f && a;
        }

        if (f){
            res.push_back(i);
        }
    }

    for (auto x : res){
        cout << x << " ";
    }
    cout << endl;



    return 0;
}